diff --git a/serial_reader_imp.h b/serial_reader_imp.h
index 9f5fd54..16e30b1 100644
--- a/serial_reader_imp.h
+++ b/serial_reader_imp.h
@@ -40,7 +40,7 @@ extern "C" {
 /* This is used to enable the Gen2 secure readdata option */
 
 #ifndef BARE_METAL
-  bool isSecureAccessEnabled ;
+  extern bool isSecureAccessEnabled ;
 #else
   static bool isSecureAccessEnabled;	
 #endif
diff --git a/serial_reader_l3.c b/serial_reader_l3.c
index 99986bf..71c4192 100644
--- a/serial_reader_l3.c
+++ b/serial_reader_l3.c
@@ -36,6 +36,10 @@
 #include "serial_reader_imp.h"
 #include "tmr_utils.h"
 
+#ifndef BARE_METAL
+bool isSecureAccessEnabled;
+#endif
+
 void
 notify_read_listeners(TMR_Reader *reader, TMR_TagReadData *trd);
 #ifdef TMR_ENABLE_SERIAL_READER
