diff --git a/tm_reader_async.c b/tm_reader_async.c
index 4bcec6f..a464c5f 100644
--- a/tm_reader_async.c
+++ b/tm_reader_async.c
@@ -40,6 +40,7 @@ TMR_Status restart_reading(struct TMR_Reader *reader);
 #include <semaphore.h>
 #include <time.h>
 #include <stdio.h>
+#include <string.h>
 
 #ifndef WIN32
 #include <sys/time.h>
diff --git a/tmr_utils.c b/tmr_utils.c
index 194a63a..80c7ce9 100644
--- a/tmr_utils.c
+++ b/tmr_utils.c
@@ -28,6 +28,7 @@
  */
 
 #include <stddef.h>
+#include <string.h>
 
 #include "osdep.h"
 #include "tmr_types.h"
@@ -171,7 +172,7 @@ TMR_stringCopy(TMR_String *dest, const char *src, int len)
   }
   if (dest->max > 0)
   {
-    tm_memcpy(dest->value, src, len);
+    memcpy(dest->value, src, len);
     dest->value[len] = '\0';
   }
 }
