diff --git a/serial_reader.c b/serial_reader.c
index 7001ab1..c711aac 100644
--- a/serial_reader.c
+++ b/serial_reader.c
@@ -4178,7 +4178,7 @@ TMR_SR_paramGet(struct TMR_Reader *reader, TMR_Param key, void *value)
     break;
   case TMR_PARAM_METADATAFLAG:
 	  {
-		*(uint16_t *)value = reader->userMetadataFlag;
+		*(TMR_TRD_MetadataFlag *)value = reader->userMetadataFlag;
 		break;
 	  }
 		    
