diff --git a/serial_transport_win32.c b/serial_transport_win32.c
index ce4cc46..ff007b6 100644
--- a/serial_transport_win32.c
+++ b/serial_transport_win32.c
@@ -126,7 +126,7 @@ s_receiveBytes(TMR_SR_SerialTransport *this, uint32_t length,
               uint32_t* messageLength, uint8_t* message, const uint32_t
 timeoutMs)
 {
- uint32_t readLength;
+ DWORD readLength;
  DWORD errorFlags;
  COMSTAT comStat;
  TMR_SR_SerialPortNativeContext *c;
